From 57516c8d11f54172364bcbb194903a645736e8c1 Mon Sep 17 00:00:00 2001
From: linjh <linjh@rock-chips.com>
Date: Sat, 29 Sep 2012 16:48:38 +0800
Subject: [PATCH] rk2928-phonepad: update rda power control

[reference file]

	modify file:
		drivers/net/wireless/rda5990/rda_5990_power_ctrl/rda_5990_power_ctrl.c
		drivers/net/wireless/rda5990/rda_wlan/rda5890_if_sdio.c
---
 .../rda_5990_power_ctrl/rda_5990_power_ctrl.c      | 14 ++++++++------
 .../wireless/rda5990/rda_wlan/rda5890_if_sdio.c    |  4 ++--
 2 files changed, 10 insertions(+), 8 deletions(-)

diff --git a/drivers/net/wireless/rda5990/rda_5990_power_ctrl/rda_5990_power_ctrl.c b/drivers/net/wireless/rda5990/rda_5990_power_ctrl/rda_5990_power_ctrl.c
index 250d032b94ae..82e0c018e7ee 100755
--- a/drivers/net/wireless/rda5990/rda_5990_power_ctrl/rda_5990_power_ctrl.c
+++ b/drivers/net/wireless/rda5990/rda_5990_power_ctrl/rda_5990_power_ctrl.c
@@ -1592,13 +1592,13 @@ static int rda_5990_pw_ioctl(struct file *file, unsigned int cmd, unsigned long
 	switch(cmd)
 	{
 		case RDA_WIFI_POWER_ON_IOCTL:
-			rda_wifi_power_on();
-			rk29sdk_wifi_set_carddetect(1);
+			//rda_wifi_power_on();
+			//rk29sdk_wifi_set_carddetect(1);
 			break;
 
 		case RDA_WIFI_POWER_OFF_IOCTL:
-		    rk29sdk_wifi_set_carddetect(0);
-			rda_wifi_power_off();
+		    	//rk29sdk_wifi_set_carddetect(0);
+			//rda_wifi_power_off();
 			break;
 
 		case RDA_WIFI_POWER_SET_TEST_MODE_IOCTL:
@@ -1658,9 +1658,10 @@ static int rda_5990_pw_ioctl(struct file *file, unsigned int cmd, unsigned long
 
 	return ret;
 }	
-
+extern int rk29sdk_wifi_power(int on);
 void mmc_rescan_slot(int id)
 {
+	rk29sdk_wifi_power(1);
     rda_wifi_power_on();
     rk29sdk_wifi_set_carddetect(1);
 }
@@ -1669,7 +1670,8 @@ EXPORT_SYMBOL(mmc_rescan_slot);
 void mmc_remove(int id)
 {
     rk29sdk_wifi_set_carddetect(0);
-			rda_wifi_power_off();
+	rda_wifi_power_off();
+	rk29sdk_wifi_power(0);
 }
 EXPORT_SYMBOL(mmc_remove);
 
diff --git a/drivers/net/wireless/rda5990/rda_wlan/rda5890_if_sdio.c b/drivers/net/wireless/rda5990/rda_wlan/rda5890_if_sdio.c
index ff738d2f17e5..44c780412897 100755
--- a/drivers/net/wireless/rda5990/rda_wlan/rda5890_if_sdio.c
+++ b/drivers/net/wireless/rda5990/rda_wlan/rda5890_if_sdio.c
@@ -37,13 +37,13 @@ extern void rk29_sdio_irq_enable(int enable);
 void export_wifi_eirq_enable(void)
 {
 	// todo...
-	rk29_sdio_irq_enable(1);
+	//rk29_sdio_irq_enable(1);
 }
 
 void export_wifi_eirq_disable(void)
 {
 	// todo...
-	rk29_sdio_irq_enable(0);
+	//rk29_sdio_irq_enable(0);
 }
 
 /* Module parameters */
-- 
2.35.3

